smart-contracts/test/Lock/partialWithdraw.js

```diff
@@ -0,0 +1,79 @@
+
+const Units = require('ethereumjs-units')
+
+const deployLocks = require('../helpers/deployLocks')
+const Unlock = artifacts.require('../Unlock.sol')
+let unlock, locks
+
+contract('Lock', (accounts) => {
+  before(async () => {
+    unlock = await Unlock.deployed()
+    locks = await deployLocks(unlock)
+  })
+
+  describe('partialWithdraw', () => {
+    let owner, price, initialLockBalance, withdrawalAmount
+    price = Units.convert('0.01', 'eth', 'wei')
+    withdrawalAmount = Units.convert('0.005', 'eth', 'wei')
+
+    before(() => {
+      const purchases = [accounts[1], accounts[2]].map((account) => {
+        return locks['OWNED'].purchaseFor(account, '', {
+          value: price,
+          from: account
+        })
+      })
+      return Promise.all(purchases)
+        .then(() => {
+          return locks['OWNED'].owner.call()
+        })
+        .then((_owner) => {
+          owner = _owner
+        })
+    })
+
+    it('should only allow the owner to withdraw', () => {
+      assert.notEqual(owner, accounts[1]) // Making sure
+      return locks['OWNED'].partialWithdraw(withdrawalAmount, {
+        from: accounts[1]
+      }).catch(error => {
+        assert.equal(error.message, 'VM Exception while processing transaction: revert')
+      })
```

Comment:
This test will not fail if it did not revert, as expected.  It will only fail if it reverts for the wrong reason.

There are various ways to handle this.  Longer term, I suggest we add a `shouldFail` helper method.  However a quick fix may be:

```
}).catch(error => {
    assert.equal(error.message, 'VM Exception while processing transaction: revert')
    return
})
assert.fail()
```

Reply:
Same comment for should fail below.

Reply:
I like the idea of a helper method, something I've used in the past as well. As for your quick fix suggestion above, eslint strips out the `return` and says the `assert.fail()` is unreachable

Reply:
Remove the previous instance of `return`

Reply:
So, stripping out the previous instance of "return" has no effect here. This seems to work, but I need another set of eyes to take a look. 
```
it('should fail if called by address other than owner', async function () {
      try {
        assert.notEqual(owner, accounts[1]) // Making sure
        await locks['OWNED'].partialWithdraw(withdrawalAmount, {
          from: accounts[1]
        })
      } catch (error) {
        assert.equal(error.message, 'VM Exception while processing transaction: revert')
        return
      }
      assert.fail()
    })
```

There are probably many places in our tests where this needs to be updated, so I'm proposing that we move ahead with a separate PR to implement a helper for testing failing transactions, possible based on this approach here which I've used in the past:
https://gist.github.com/xavierlepretre/d5583222fde52ddfbc58b7cfa0d2d0a9

Reply:
Yea that looks good now.  

I'd love to see the helper added.  I've previously noticed other 'should fail' tests which don't fail when no revert occurs (e.g. all the `approve` should fail tests).  The helper along with async style could make our tests much more readable as well!